Mixpanel/MixpanelNotificationServiceExtension.swift

```diff
@@ -0,0 +1,168 @@
+import UserNotifications
+
+@available(iOS 11.0, *)
+open class MixpanelNotificationServiceExtension: UNNotificationServiceExtension {
+
+    var contentHandler: ((UNNotificationContent) -> Void)?
+    var bestAttemptContent: UNMutableNotificationContent?
+
+    open override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+
+        NSLog("%@ MPNotificationServiceExtension didReceiveNotificationRequest", self);
+
+        guard MixpanelPushNotifications.isMixpanelPushNotification(request.content) else {
+            NSLog("%@ Not a Mixpanel push notification, returning original content", self);
```

Comment:
Can we use Logger.info in here?

Reply:
No, my understanding is that this is a completely new environment which means it's a different Mixpanel instance. Thus, we'd have to require customers to explicitly create a Mixpanel instance here and enable logging to get these. My thoughts were that this is a pretty rare thing so always `NSLog`ing won't pollute the logs much.